app/controllers/component-tree.js

```diff
@@ -1,314 +1,358 @@
+    renderNode.type === 'route-template' && renderNode.name === '-top-level';
+}
+
+class RenderItem {
+  @tracked isExpanded = true;
+
+  constructor(controller, parentItem, renderNode) {
+    this.controller = controller;
+    this.parentItem = parentItem;
+    this.renderNode = renderNode;
+  }
+
+  get id() {
+    return this.renderNode.id;
+  }
+
+  get isOutlet() {
+    return this.renderNode.type === 'outlet';
+  }
+
+  get isEngine() {
+    return this.renderNode.type === 'engine';
+  }
+
+  get isRouteTemplate() {
+    return this.renderNode.type === 'route-template';
+  }
+
+  get isComponent() {
+    return this.renderNode.type === 'component';
+  }
+
+  get name() {
+    return this.renderNode.name;
+  }
+
+  get hasInstance() {
+    let { instance } = this.renderNode;
+    return typeof instance === 'object' && instance !== null;
+  }
+
+  get instance() {
+    if (this.hasInstance) {
+      return this.renderNode.instance.id;
+    } else {
+      return null;
+    }
+  }
+
+  get hasBounds() {
+    return this.renderNode.bounds !== null;
+  }
+
+  get isRoot() {
+    return this.parentItem === null;
+  }
+
+  get level() {
+    if (this.isRoot) {
+      return 0;
+    } else {
+      return this.parentItem.level + 1;
+    }
+  }
+
+  get hasChildren() {
+    return this.childItems.length > 0;
+  }
+
+  get childItems() {
+    return this.renderNode.children
+      .filter(child => !isInternalRenderNode(child))
+      .map(child => this.controller.findItem(child.id));
+  }
+
+  get isVisible() {
+    if (this.isRoot) {
+      return true;
+    } else {
+      return this.parentItem.isVisible && this.parentItem.isExpanded;
+    }
+  }
+
+  get isSelected() {
+    return this.id === this.controller.pinned;
+  }
+
+  get isHighlighted() {
+    return this.id === this.controller.previewing;
+  }
+
+  get style() {
+    let indentation = 25;
+
+    indentation += this.level * 20;
+
+    if (this.hasChildren) {
+      // folding triangle
+      indentation -= 12;
+    }
+
+    return htmlSafe(`padding-left: ${indentation}px`);
+  }
+
+  @action showPreview() {
+    this.controller.previewing = this.id;
+  }
+
+  @action hidePreview() {
+    if (this.isHighlighted) {
+      this.controller.previewing = undefined;
+    }
+  }
+
+  @action toggleInspection() {
+    if (this.isSelected) {
+      this.controller.pinned = undefined;
+    } else {
+      this.controller.pinned = this.id;
+    }
+  }
+
+  @action toggleExpansion(event) {
     event.stopPropagation();
 
-    if (event.altKey) {
-      this.toggleWithChildren(item);
+    if (this.isExpanded) {
+      this.collapse(event.altKey);
     } else {
-      item.toggleProperty('expanded');
-      this.expandedStateCache[getIdFromObj(item)] = item.get('expanded');
+      this.expand(event.altKey);
     }
-  }),
+  }
 
-  viewInElementsPanel: action(function(item, event) {
+  @action scrollIntoView() {
     event.stopPropagation();
```

Comment:
Hmm, where does `event` come from? It used to be passed as an action argument is that still happening? I don’t see where `event` is defined at all (why isn’t linting failing?).